tests/lib/dependencies.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+# shellcheck source=tests/lib/pkgdb.sh
+. "$TESTSLIB/pkgdb.sh"
+
+export DEPENDENCY_PACKAGES=
+export DISTRO_BUILD_DEPS=
+
+add_pkgs(){
+  DEPENDENCY_PACKAGES="$DEPENDENCY_PACKAGES $@"
+}
+
+get_apt_dependencies_generic(){
+  add_pkgs autoconf
```

Comment:
Communicating across shell functions via sticky environment variables is pretty hard to follow. It's a backdoor without a clear flow between the involved parties. Instead, this can simply `echo autoconf` here, and the other side can use the standard output as usual for its needs.